### ISSUE TITLE
My selector-class-pattern freezes stylelint in some edge cases

### ISSUE THREAD

**Issue by silvenon** [stylelint/stylelint#848]
From AtomLinter/linter-stylelint#106.

> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

My particular `selector-class-pattern` rule freezes stylelint.

> Which rule, if any, is this issue related to?

`selector-class-pattern`

> What CSS is needed to reproduce this issue?

```css
/* removed parenthesis is intentional, this is what's freezing the process */
:global.foo) {
  color: pink;
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "selector-class-pattern": "^[a-z]+([A-Z0-9][a-z0-9]+[A-Za-z0-9]*)?$",
  }
}
```

> Which version of stylelint are you using?

`4.4.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI.

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No.

> What did you expect to happen?

For stylelint not to freeze :smile: 

> What actually happened (e.g. what warnings or errors you are getting)?

It froze.

**Comment by davidtheclark** [comment]
By "in some edge cases" you meant that it doesn't freeze every time? Any patterns you can see?

**Comment by silvenon** [comment]
@davidtheclark I meant that `:global.foo)` is an edge case (it freezes every time). It doesn't freeze if I do `:global(.foo` or `.fooBar`. There is also one edge case I found with square brackets, I'll edit my post if I remember.

**Comment by jeddy3** [comment]
This is being discussed upstream in https://github.com/postcss/postcss-selector-parser/pull/43

**Comment by ben-eb** [comment]
postcss-selector-parser 1.3.3 will now throw an error instead of going into an infinite loop.

**Comment by jeddy3** [comment]
@ben-eb A big thank you for resolving this so quickly!

**Comment by ben-eb** [comment]
You're welcome. :smile:

**Comment by davidtheclark** [comment]
Nice job fellas.

**Comment by silvenon** [comment]
:+1:
